dagster_sqlmesh/translator.py

```diff
@@ -1,27 +1,34 @@
+import re
+from collections.abc import Sequence
 
-import sqlglot
 from dagster import AssetKey
-from sqlglot import exp
 from sqlmesh.core.context import Context
 from sqlmesh.core.model import Model
 
 
 class SQLMeshDagsterTranslator:
     """Translates sqlmesh objects for dagster"""
 
-    def get_asset_key_from_model(self, context: Context, model: Model) -> AssetKey:
+    def get_asset_key(self, context: Context, fqn: str) -> AssetKey:
         """Given the sqlmesh context and a model return the asset key"""
-        return AssetKey(model.view_name)
-
-    def get_asset_key_fqn(self, context: Context, fqn: str) -> AssetKey:
-        """Given the sqlmesh context and a fqn of a model return an asset key"""
-        table = self.get_fqn_to_table(context, fqn)
-        return AssetKey(table.name)
-
-    def get_fqn_to_table(self, context: Context, fqn: str) -> exp.Table:
-        """Given the sqlmesh context and a fqn return the table"""
-        dialect = self._get_context_dialect(context)
-        return sqlglot.to_table(fqn, dialect=dialect)
+        path = self.get_asset_key_name(fqn)
+        return AssetKey(path)
+
+    def get_asset_key_name(self, fqn: str) -> Sequence[str]:
+        asset_path = re.findall(r"[A-Za-z0-9_\-]+", fqn)
```

Comment:
I would not parse the fqn with a regex, I'd prefer to rely on the tested parsing of sqlglot here.

I'd add this back.

```
from sqlglot import exp
```

And then make this change and just return the table expression. 

```suggestion
        table_exp = exp.to_table(fqn)
        return table_exp
```

Reply:
@ravenac95 I changed it to the table expression by parts, I think it's a good idea to show that Dagster assets can have a composite key: catalogue-source-asset, and if the end user wants to change it, they can always do that. But do let me know if you don't agree, we can always change this.